### ISSUE TITLE
genieparser/src/genie/libs/parser/iosxe /show_interface.py

### ISSUE THREAD

**Issue by dmychepk** [CiscoTestAutomation/genieparser#805]
parser "show interfaces" doers not return carrier transitions at the end of show command:
GigabitEthernet0/0/0 is up, line protocol is up 
  Hardware is BUILT-IN-2T+6X1GE, address is cc16.7e85.0e02 (bia cc16.7e85.0e02)
  Internet address is 152.22.242.102/26
  MTU 1500 bytes, BW 1000000 Kbit/sec, DLY 10 usec, 
     reliability 255/255, txload 1/255, rxload 1/255
  Encapsulation ARPA, loopback not set
  Keepalive not supported 
  Full Duplex, 1000Mbps, link type is auto, media type is T
  output flow-control is on, input flow-control is on
  ARP type: ARPA, ARP Timeout 04:00:00
  Last input 00:00:00, output 00:00:01, output hang never
  Last clearing of "show interface" counters never
  Input queue: 0/375/0/0 (size/max/drops/flushes); Total output drops: 0
  Queueing strategy: Class-based queueing
  Output queue: 0/40 (size/max)
  30 second input rate 28000 bits/sec, 31 packets/sec
  30 second output rate 31000 bits/sec, 14 packets/sec
     2474975 packets input, 299064260 bytes, 0 no buffer
     Received 1204070 broadcasts (0 IP multicasts)
     0 runts, 0 giants, 0 throttles 
     0 input errors, 0 CRC, 0 frame, 0 overrun, 0 ignored
     0 watchdog, 51648 multicast, 0 pause input
     1630775 packets output, 624450148 bytes, 0 underruns
     Output 401 broadcasts (0 IP multicasts)
     0 output errors, 0 collisions, 2 interface resets
     121 unknown protocol drops
     0 babbles, 0 late collision, 0 deferred
     0 lost carrier, 0 no carrier, 0 pause output
     0 output buffer failures, 0 output buffers swapped out
     **3 carrier transitions**

**Comment by SohanTirpude** [comment]
Hello @dmychepk,

Kindly share the command output and the resultant output of `parse` method with keeping formatting intact. Above shared output is not properly formatted so it is difficult to debug this issue. 

Kindly share raw output at it is and let me know.

Thank you.

**Comment by dmychepk** [comment]
```
BR-21#show int GigabitEthernet 0/0/0
GigabitEthernet0/0/0 is up, line protocol is up 
  Hardware is BUILT-IN-2T+6X1GE, address is cc16.7e85.0e02 (bia cc16.7e85.0e02)
  Internet address is 152.22.242.102/26
  MTU 1500 bytes, BW 1000000 Kbit/sec, DLY 10 usec, 
     reliability 255/255, txload 1/255, rxload 1/255
  Encapsulation ARPA, loopback not set
  Keepalive not supported 
  Full Duplex, 1000Mbps, link type is auto, media type is T
  output flow-control is on, input flow-control is on
  ARP type: ARPA, ARP Timeout 04:00:00
  Last input 00:00:00, output 00:00:02, output hang never
  Last clearing of "show interface" counters never
  Input queue: 0/375/0/0 (size/max/drops/flushes); Total output drops: 128035
  Queueing strategy: Class-based queueing
  Output queue: 0/40 (size/max)
  30 second input rate 28000 bits/sec, 29 packets/sec
  30 second output rate 50000 bits/sec, 17 packets/sec
     11059827 packets input, 1600856154 bytes, 0 no buffer
     Received 5157390 broadcasts (0 IP multicasts)
     0 runts, 0 giants, 0 throttles 
     0 input errors, 0 CRC, 0 frame, 0 overrun, 0 ignored
     0 watchdog, 232903 multicast, 0 pause input
     7598063 packets output, 2854756614 bytes, 0 underruns
     Output 1297 broadcasts (0 IP multicasts)
     0 output errors, 0 collisions, 2 interface resets
     556 unknown protocol drops
     0 babbles, 0 late collision, 0 deferred
     0 lost carrier, 0 no carrier, 0 pause output
     0 output buffer failures, 0 output buffers swapped out
     3 carrier transitions

Parser class: <class 'genie.libs.parser.iosxe.show_interface.ShowInterfaces'> arguments: {}
'GigabitEthernet0/0/0': {
        'arp_timeout': '04:00:00',
        'arp_type': 'arpa',
        'auto_negotiate': True,
        'bandwidth': 1000000,
        'counters': {
            'in_broadcast_pkts': 1459027,
            'in_crc_errors': 0,
            'in_errors': 0,
            'in_frame': 0,
            'in_giants': 0,
            'in_ignored': 0,
            'in_mac_pause_frames': 0,
            'in_multicast_pkts': 62147,
            'in_no_buffer': 0,
            'in_octets': 589581302,
            'in_overrun': 0,
            'in_pkts': 3141493,
            'in_runts': 0,
            'in_throttles': 0,
            'in_watchdog': 0,
            'last_clear': 'never',
            'out_babble': 0,
            'out_broadcast_pkts': 493,
            'out_buffer_failure': 0,
            'out_buffers_swapped': 0,
            'out_collision': 0,
            'out_deferred': 0,
            'out_errors': 0,
            'out_interface_resets': 2,
            'out_late_collision': 0,
            'out_lost_carrier': 0,
            'out_mac_pause_frames': 0,
            'out_multicast_pkts': 0,
            'out_no_carrier': 0,
            'out_octets': 757519950,
            'out_pkts': 2040070,
            'out_underruns': 0,
            'out_unknown_protocl_drops': 147,
            'rate': {
                'in_rate': 25000,
                'in_rate_pkts': 27,
                'load_interval': 30,
                'out_rate': 38000,
                'out_rate_pkts': 15,
            },
        },
        'delay': 10,
        'duplex_mode': 'full',
        'enabled': True,
        'encapsulations': {
            'encapsulation': 'arpa',
        },
        'flow_control': {
            'receive': True,
            'send': True,
        },
        'ipv4': {
            '152.22.242.102/26': {
                'ip': '152.22.242.102',
                'prefix_length': '26',
            },
        },
        'is_deleted': False,
        'last_input': '00:00:00',
        'last_output': '00:00:03',
        'line_protocol': 'up',
        'link_type': 'auto',
        'mac_address': 'cc16.7e85.0e02',
        'media_type': 'T',
        'mtu': 1500,
        'oper_status': 'up',
        'output_hang': 'never',
        'phys_address': 'cc16.7e85.0e02',
        'port_channel': {
            'port_channel_member': False,
        },
        'port_speed': '1000mbps',
        'queues': {
            'input_queue_drops': 0,
            'input_queue_flushes': 0,
            'input_queue_max': 375,
            'input_queue_size': 0,
            'output_queue_max': 40,
            'output_queue_size': 0,
            'queue_strategy': 'Class-based',
            'total_output_drop': 0,
        },
        'reliability': '255/255',
        'rxload': '1/255',
        'txload': '1/255',
        'type': 'BUILT-IN-2T+6X1GE',
    }

**Comment by SohanTirpude** [comment]
Hello @dmychepk, 

The PR has been raised for this. Once it is merged, I will update you. 

Thank you

**Comment by SohanTirpude** [comment]
Hello @dmychepk,

The PR has been merged and the fix will be available in the upcoming pyATS version. This is how the output now looks like: 

```
{
    "GigabitEthernet0/0/0": {
        "port_channel": {
            "port_channel_member": False
        },
        "is_deleted": False,
        "enabled": True,
        "line_protocol": "up",
        "oper_status": "up",
        "type": "BUILT-IN-2T+6X1GE",
        "mac_address": "cc16.7e85.0e02",
        "phys_address": "cc16.7e85.0e02",
        "ipv4": {
            "152.22.242.102/26": {
                "ip": "152.22.242.102",
                "prefix_length": "26"
            }
        },
        "delay": 10,
        "mtu": 1500,
        "bandwidth": 1000000,
        "reliability": "255/255",
        "txload": "1/255",
        "rxload": "1/255",
        "encapsulations": {
            "encapsulation": "arpa"
        },
        "duplex_mode": "full",
        "port_speed": "1000mbps",
        "link_type": "auto",
        "auto_negotiate": True,
        "media_type": "T",
        "flow_control": {
            "receive": True,
            "send": True
        },
        "arp_type": "arpa",
        "arp_timeout": "04:00:00",
        "last_input": "00:00:00",
        "last_output": "00:00:02",
        "output_hang": "never",
        "queues": {
            "input_queue_size": 0,
            "input_queue_max": 375,
            "input_queue_drops": 0,
            "input_queue_flushes": 0,
            "total_output_drop": 128035,
            "queue_strategy": "Class-based",
            "output_queue_size": 0,
            "output_queue_max": 40
        },
        "counters": {
            "rate": {
                "load_interval": 30,
                "in_rate": 28000,
                "in_rate_pkts": 29,
                "out_rate": 50000,
                "out_rate_pkts": 17
            },
            "last_clear": "never",
            "in_pkts": 11059827,
            "in_octets": 1600856154,
            "in_no_buffer": 0,
            "in_multicast_pkts": 232903,
            "in_broadcast_pkts": 5157390,
            "in_runts": 0,
            "in_giants": 0,
            "in_throttles": 0,
            "in_errors": 0,
            "in_crc_errors": 0,
            "in_frame": 0,
            "in_overrun": 0,
            "in_ignored": 0,
            "in_watchdog": 0,
            "in_mac_pause_frames": 0,
            "out_pkts": 7598063,
            "out_octets": 2854756614,
            "out_underruns": 0,
            "out_broadcast_pkts": 1297,
            "out_multicast_pkts": 0,
            "out_errors": 0,
            "out_interface_resets": 2,
            "out_collision": 0,
            "out_unknown_protocl_drops": 556,
            "out_babble": 0,
            "out_late_collision": 0,
            "out_deferred": 0,
            "out_lost_carrier": 0,
            "out_no_carrier": 0,
            "out_mac_pause_frames": 0,
            "out_buffer_failure": 0,
            "out_buffers_swapped": 0
        },
        "carrier_transitions": 3
    }
}
```

I hope it is meeting your expectations. I will be closing this issue for now, if even with newer pyATS release you are not able to see this change, then kindly re-open this ticket.

Thank you.
